meson.build

```diff
@@ -0,0 +1,237 @@
+if harfbuzz_dep.found()
+    deps += harfbuzz_dep
+    conf.set('CONFIG_HARFBUZZ', 1)
+endif
+
+fontconfig_dep = dependency('fontconfig', version: '>= 2.10.92', required: get_option('fontconfig'))
+fontconfig = fontconfig_dep.found()
+if fontconfig
+    deps += fontconfig_dep
+    conf.set('CONFIG_FONTCONFIG', 1)
+endif
+
+if get_option('test')
+    deps += dependency('libpng', version: '>= 1.2.0')
+    conf.set('CONFIG_LIBPNG', 1)
+endif
+
+# DirectWrite
+directwrite = false
+if host_machine.system() == 'windows' and not get_option('directwrite').disabled()
+    if cc.has_header('windows.h')
+        deps += cc.find_library('dwrite', required: true)
+        conf.set('CONFIG_DIRECTWRITE', 1)
+        directwrite = true
+    elif get_option('directwrite').enabled()
+        error('DirectWrite was enabled explicitly, but required header is missing.')
+    endif
+endif
+
+# CoreText
+coretext = false
+if host_machine.system() == 'darwin' and not get_option('coretext').disabled()
+    coretext_dep = dependency('appleframeworks', modules: ['CoreText', 'CoreFoundation'], required: false)
+    if cc.has_type('CTRunRef', prefix: '#include <CoreText/CoreText.h>', dependencies: coretext_dep)
+        deps += coretext_dep
+        conf.set('CONFIG_CORETEXT', 1)
+        coretext = true
+    elif get_option('coretext').enabled()
+        error('CoreText was enabled explicitly, but required headers or frameworks are missing.')
+    endif
+endif
+
+if get_option('require-system-font-provider') and not fontconfig and not directwrite and not coretext
+    error('''Either DirectWrite (on Windows), CoreText (on OSX), or Fontconfig (Linux, other) is required.
+If you really want to compile without a system font provider, set -Drequire-system-font-provider=false''')
+endif
+
+# ASM
+
+nasm_args = []
+nasm = find_program('nasm', required: get_option('asm'))
+enable_asm = nasm.found()
+
+# NASM version check
+if enable_asm
+    nasm_r = run_command(nasm, '-v', check: true)
+    out = nasm_r.stdout().strip().split()
+    if out[1].to_lower() == 'version'
+        if out[2].version_compare('< 2.10')
+            warning('nasm is too old (found @0@); ASM functions are disabled.'.format(out[2]))
+            enable_asm = false
+        endif
+    else
+        warning('Unexpected nasm version string: @0@; disabling.'.format(nasm_r.stdout()))
+        enable_asm = false
+    endif
+endif
+
+# Architecture check - TODO x32
+if host_machine.cpu_family() == 'x86'
+    bittype = '32'
+    nasm_args += '-DARCH_X86_64=0'
+    x86 = true
+elif host_machine.cpu_family() == 'x86_64'
+    bittype = '64'
+    nasm_args += '-DARCH_X86_64=1'
+    x86 = true
+else
+    x86 = false
+endif
+
+if enable_asm and not x86
+    warning('ASM functions are not available on non-x86 machines; disabling.')
+    enable_asm = false
+elif (not enable_asm) and x86
+    warning('Install nasm-2.10 or later for a significantly faster libass build.')
+endif
```

Comment:
If assembly was explicitly requested but is unavailable this should error out.

Reply:
Checking for the nasm program uses a feature check, so it has enabled/disabled/auto and if enabled, it will error out when not found.

So if it was enabled and found, but the version was insufficient or not implemented for the current platform, it should indeed error out, i.e. the first warning.

The second warning is only triggered when asm was not explicitly requested, but instead set to "auto" (check and graceful fallback), in which case a non-fatal warning is appropriate... or if it was set to disabled, in which case since the user specifically asked not to use it even if nasm is installed, in which case it may make sense to be silent and not warn.

Reply:
There are three possible statuses for a user request, and each deserves a different severity.

Currently, the PR only performs the "warn" severity. (i.e. "the user didn't say whether they want it, so guess what to do", in autotools parlance, `--with-asm=check`).

You pointed out that if the user explicitly requests it (in autotools parlance, `--with-asm=yes`) it should be an error. I agreed with you.

I pointed out that there's also a third case, where the user explicitly says "do not use it" (in autotools parlance, `--with-asm=no`) which should be silent.